### PR TITLE
Mapping local absolute file paths

### PR DESCRIPTION
**Pull request by AndrewNatoli**
As mentioned in #11...

This will resolve the issue with mapping absolute file paths with the recent sourcemap additions.

If an absolute path is used, when determining the local file to open the adapter will replace the `serverSourceRoot` path with the `localSourceRoot`.

So if `serverSourceRoot` is /var/www/ and `localSourceRoot` is /Users/andrew/project/, we will take /var/www/lol.php and change it to /Users/andrew/project/lol.php.

The original logic to map relative local paths is preserved and will be used if the `localSourceRoot` begins with `.`. 

Additionally, if `serverSourceRoot` ends with a trailing slash, a trailing slash will be added to `localSourceRoot` and vice-versa to make the string.replace more convenient. 

This also ensures that `serverFileUri` is prepended with "file://" before it's sent with `set_breakpoint`.

Lastly, when parsing the uri received from XDebug (which looks like file://path/to/file.php), the original logic of removing the first character after file:// is preserved on Windows, but on non-windows systems we won't remove the first character. This will preserve /paths/like/this/.

------

Tested with....
- Remote mapping (Local = Mac, Remote = Linux)
- Local mapping
- Local (without mapping)

### CODE REVIEW

File: src/phpDebug.ts [hunk 1]
```diff
@@ -197,14 +213,20 @@ class PhpDebugSession extends vscode.DebugSession {
 
     /** converts a server-side XDebug file URI to a local path for VS Code with respect to source root settings */
     protected convertDebuggerPathToClient(fileUri: string): string {
-        // convert the file URI to a path
-        const serverPath = decodeURI(url.parse(fileUri).pathname.substr(1));
+        // convert the file URI to a path. Don't remove starting slash on unix platforms.
+        let n:number = (/^win/.test(process.platform)) ? 1 : 0;
+        const serverPath = url.parse(fileUri).pathname.substr(n);
```
Comment (felixfbecker) [inline src/phpDebug.ts]: This is great in general, but a check for `process.platform === 'win32'` is enough (it has nothing to do with 32 / 64 bit)

Comment (AndrewNatoli) [inline src/phpDebug.ts]: Okay, I'll get this line revised.

**Comment by felixfbecker** [comment]
I squashed my commits before I merged sourcemaps, seems like you based on my old commits. You will have to rebase interactively on upstream/master and throw my commits away. Sorry for the inconvenience.

I will revisit this tomorrow, way too tired right now (3 AM here)

**Comment by felixfbecker** [comment]
In general, always start PRs of from master, not a feature branch. That feature might in the end not get merged. If it does, you can rebase before submitting ;)

**Comment by felixfbecker** [comment]
And I'm a bit on the fence with this. My inital implementation used RegExps and in the end I opted for url/path resolve functions because they take care of all the inconsistencies like trailing slashes or windows drive letters in lowercase / uppercase by themselves. I would prefer if you find a solution to your issue with these tools insteads.

**Comment by AndrewNatoli** [comment]
As you might be able to tell, an adventure happened during that rebase.

**Pull request closed without merging**